dotcom-rendering/src/server/render.hostedContent.web.tsx

```diff
@@ -0,0 +1,56 @@
+import { isString } from '@guardian/libs';
+import { HostedArticleLayout } from '../layouts/HostedArticleLayout';
+import { HostedGalleryLayout } from '../layouts/HostedGalleryLayout';
+import { getModulesBuild, getPathFromManifest } from '../lib/assets';
+import { renderToStringWithEmotion } from '../lib/emotion';
+import { polyfillIO } from '../lib/polyfill.io';
+import type { HostedContent } from '../types/hostedContent';
+import { htmlPageTemplate } from './htmlPageTemplate';
+
+type Props = {
+	hostedContent: HostedContent;
+};
+
+export const renderHtml = ({ hostedContent }: Props) => {
+	const { type, frontendData } = hostedContent;
+
+	const title = `Advertiser content hosted by the Guardian: ${frontendData.title} | The Guardian`;
+
+	const HostedLayout =
+		type === 'gallery' ? HostedGalleryLayout : HostedArticleLayout;
+	const renderingTarget = 'Web';
+
+	const { html, extractedCss } = renderToStringWithEmotion(
+		<HostedLayout renderingTarget={renderingTarget} />,
+	);
+
+	// We don't send A/B tests or switches from frontend yet- do we need to?
+	const build = getModulesBuild({
+		tests: {},
+		switches: {},
+	});
+
+	const prefetchScripts = [
+		polyfillIO,
+		getPathFromManifest(build, 'frameworks.js'),
+		getPathFromManifest(build, 'index.js'),
+	].filter(isString);
+
+	// We currently don't send any of the data required for page config or window.guardian setup from frontend
+	const pageHtml = htmlPageTemplate({
+		scriptTags: [],
+		css: extractedCss,
+		html,
+		title,
+		description: frontendData.standfirst,
+		// @ts-expect-error no config data
+		guardian: {},
```

Comment:
Based on the `createGuardian` function is it best to populate the `HostedContent["frontendData"]` object with the `Config` object, as most of the fields like `sentryHost`, `frontendAssetsFullURL`, `googletagUrl` are needed for the pages? Or would you recommend we wait for the full spec of the page to be released before attempting to populate the `window.guardian` object?

Reply:
We need the information to be passed from frontend, we can't do anything about this in DCR as DCR doesn't currently detemine any of this information. I don't know what you mean by 'waiting for the full spec of the page to be released'; we are in charge of this work and will need to make the necessary changes in frontend.